src/HotChocolate/Core/src/Execution/Pipeline/OperationExecutionMiddleware.cs

```diff
@@ -103,16 +105,41 @@ internal sealed class OperationExecutionMiddleware
                                 mutation,
                                 context.Variables);
 
-                            context.Result = await _mutationExecutor
+                            result = await _mutationExecutor
                                 .ExecuteAsync(operationContext)
                                 .ConfigureAwait(false);
                         }
 
+                        if (operationContext.Execution.DeferredTaskBacklog.IsEmpty || 
+                            result is null)
```

Comment:
can the inital result of defere be null?

Reply:
theoretically, if a new operation is introduced, in this case the result will be null and we yield to the next middleware.